### PR TITLE
add line, col to error message

### PR DESCRIPTION
**Pull request by denar90**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -99,6 +99,11 @@ class StylusCompiler {
       });
     });
   }
+
+  _getError(error) {
+    const re = /:(\d+):(\d+)/.exec(error.message);
```
Comment (shvaikalesh) [inline index.js]: Let's make it `/(:\d+){2}/`

File: index.js [hunk 2]
```diff
@@ -89,7 +89,7 @@ class StylusCompiler {
 
     return new Promise((resolve, reject) => {
       compiler.render((error, data) => {
-        if (error) return reject(error);
+        if (error) return reject(this._getError(error));
```
Comment (shvaikalesh) [inline index.js]: I believe this should not be a method, just a helper.

File: index.js [hunk 3]
```diff
@@ -99,6 +99,11 @@ class StylusCompiler {
       });
     });
   }
+
+  _getError(error) {
+    const re = /:(\d+):(\d+)/.exec(error.message);
+    return new Error(`:${re[1]}:${re[2]} ${error.message}`);
```
Comment (shvaikalesh) [inline index.js]: Can we just reject with a string?

File: index.js (hunk 1 above)
Comment (denar90) [inline index.js]: But we will not achieve line... 
https://regex101.com/r/EfqmIf/1

File: index.js [hunk 4]
```diff
@@ -23,6 +23,8 @@ const cssModulify = (path, data, map) => {
   });
 };
 
+const getError = error => `${/(:\d+){2}/.exec(error.message)[0]} ${error.message}`;
```
Comment (shvaikalesh) [inline index.js]: Maybe we should move regexp out of `getError` and call it like `loc/location`?

File: index.js [hunk 5]
```diff
@@ -23,6 +23,9 @@ const cssModulify = (path, data, map) => {
   });
 };
 
+const location = message => /(:\d+){2}/.exec(message)[0];
+const getError = error => `${location(error.message)} ${error.message}`;
```
Comment (shvaikalesh) [inline index.js]:
```
`${location(error.message)} ${error}`;
```

File: test.js [hunk 6]
```diff
@@ -56,6 +56,18 @@ describe('Plugin', () => {
         expect(result.data).to.equal(expected);
       });
     });
+
+    it('should throw error in correct format', () => {
+      const data = ">";
+      const expected = ':1:2 ParseError: fixtures/app/styles/style.styl:1:2\n' +
```
Comment (shvaikalesh) [inline test.js]: template literal

**Comment by shvaikalesh** [comment]
@paulmillr this one is not ready yet

**Pull request merged**